spring-boot-project/spring-boot-docs/src/main/asciidoc/spring-boot-features.adoc

```diff
@@ -5622,8 +5622,19 @@ the auto-configured `KafkaTemplate`.
 When the Apache Kafka infrastructure is present, any bean can be annotated with
 `@KafkaListener` to create a listener endpoint. If no `KafkaListenerContainerFactory` has
 been defined, a default one is automatically configured with keys defined in
-`spring.kafka.listener.*`. Also, if a `RecordMessageConverter` bean is defined, it is
-automatically associated to the default factory.
+`spring.kafka.listener.*`.
+If the property `spring.kafka.producer.transaction-id-prefix` is defined, a
+`KafkaTransactionManager` will be auto-configured with name `kafkaTransactionManager` and
```

Comment:
Where is the behaviour implemented? All I can see here is that a `TransactionManager` is associated if it exists. Is `@EnableKafka` doing that for us? Is that a `PlatformTransactionManager` (in which case we need to be careful about conflicts).

Reply:
A TM is already autoconfigured in `KafkaAutoConfiguration`, but I guess it was never documented (looks like a community contribution).

```java
	@Bean
	@ConditionalOnProperty(name = "spring.kafka.producer.transaction-id-prefix")
	@ConditionalOnMissingBean
	public KafkaTransactionManager<?, ?> kafkaTransactionManager(
			ProducerFactory<?, ?> producerFactory) {
		return new KafkaTransactionManager<>(producerFactory);
	}
```